integration-tests/testsetups/ocr.go

```diff
@@ -655,22 +692,54 @@ func (o *OCRSoakTest) complete() {
 	o.TestReporter.RecordEvents(o.ocrRoundStates, o.testIssues)
 }
 
-func (o *OCRSoakTest) startBlockchainReorg(depth int) {
-	if !o.Config.GetNetworkConfig().IsSimulatedGethSelected() {
-		require.FailNow(o.t, "Reorg only available for Simulated Geth")
-		return
-	}
-
-	client := ctf_client.NewRPCClient(o.rpcNetwork.HTTPURLs[0])
+func (o *OCRSoakTest) startGethBlockchainReorg(network blockchain.EVMNetwork, conf ctf_config.ReorgConfig) {
+	client := ctf_client.NewRPCClient(network.HTTPURLs[0])
 	o.log.Info().
 		Str("URL", client.URL).
-		Int("depth", depth).
+		Int("Depth", conf.Depth).
 		Msg("Starting blockchain reorg on Simulated Geth chain")
-	err := client.GethSetHead(depth)
+	err := client.GethSetHead(conf.Depth)
 	require.NoError(o.t, err, "Error starting blockchain reorg on Simulated Geth chain")
 	o.reorgHappened = true
 }
 
+func (o *OCRSoakTest) startAnvilGasSpikeSimulation(network blockchain.EVMNetwork, conf ctf_config.GasSpikeSimulationConfig) {
+	client := ctf_client.NewRPCClient(network.HTTPURLs[0])
+	o.log.Info().
+		Str("URL", client.URL).
+		Any("GasSpikeSimulationConfig", conf).
+		Msg("Starting gas spike simulation on Anvil chain")
+	err := client.ModulateBaseFeeOverDuration(o.log, conf.StartGasPrice, conf.GasRisePercentage, conf.Duration.Duration, conf.GasSpike)
+	require.NoError(o.t, err, "Error starting gas simulation on Anvil chain")
+	o.gasSpikeSimulationHappened = true
+}
+
+func (o *OCRSoakTest) startAnvilGasLimitSimulation(network blockchain.EVMNetwork, conf ctf_config.GasLimitSimulationConfig) {
+	client := ctf_client.NewRPCClient(network.HTTPURLs[0])
+	latestBlock, err := o.seth.Client.BlockByNumber(context.Background(), nil)
+	require.NoError(o.t, err)
+	newGasLimit := int64(math.Ceil(float64(latestBlock.GasUsed()) * conf.NextGasLimitPercentage))
+	o.log.Info().
+		Str("URL", client.URL).
+		Any("GasLimitSimulationConfig", conf).
+		Uint64("LatestBlock", latestBlock.Number().Uint64()).
+		Uint64("LatestGasUsed", latestBlock.GasUsed()).
+		Uint64("LatestGasLimit", latestBlock.GasLimit()).
+		Int64("NewGasLimit", newGasLimit).
+		Msg("Starting new gas limit simulation on Anvil chain")
+	err = client.AnvilSetBlockGasLimit([]interface{}{newGasLimit})
+	require.NoError(o.t, err, "Error starting gas simulation on Anvil chain")
+	time.Sleep(conf.Duration.Duration)
```

Comment:
curious, why we need to sleep here?

Reply:
@Tofel this is to wait before returning to the old gas limit in the next lines